pkg/scheduler/placement/fixed_rule.go

```diff
@@ -121,12 +121,19 @@ func (fr *fixedRule) placeApplication(app *objects.Application, queueFn func(str
 		if parentName == "" {
 			parentName = configs.RootQueue
 		}
-		queueName = parentName + configs.DOT + fr.queue
+		childQueueName := replaceDot(fr.queue)
+		if err = configs.IsQueueNameValid(childQueueName); err != nil {
+			return "", err
+		}
+		queueName = parentName + configs.DOT + childQueueName
 	}
 	// Log the result before we check the create flag
 	log.Log(log.SchedApplication).Debug("Fixed rule intermediate result",
 		zap.String("application", app.ApplicationID),
 		zap.String("queue", queueName))
+	if err := configs.IsQueuePathValid(queueName); err != nil {
+		return "", err
+	}
```

Comment:
If each part is valid the path must be valid. We should have checked each part before we get to the final queue path assembly.
At this point we cannot distinguish anymore between valid and invalid `.` dot characters so it adds nothing.

Reply:
Did not repeat the comment for every rule this was part of.